sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/clienttelemetry/ClientTelemetry.java

```diff
@@ -212,12 +212,14 @@ public boolean isClientMetricsEnabled() {
         return this.clientMetricsEnabled;
     }
 
-    public void init() {
-        loadAzureVmMetaData();
+    public Mono<?> init() {
+        return loadAzureVmMetaData()
+            .doOnTerminate(() -> {
```

Comment:
Instead of `doOnTerminate()` here, we should use `then()` operator, to avoid subscribing another flux/mono inside this Mono. 
We can then just `return sendClientTelemetry()` - as the outer method is now already subscribing to this `init()` function.

Reply:
Resolved